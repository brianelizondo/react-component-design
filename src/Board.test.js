import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Board from "./Board";

// Smoke Tests
it("'Board' component renders without crashing", function() {
    render(<Board />);
});

// Snapshot Tests
it("'Board' matches snapshot with light start off", function() {
    const {asFragment} = render(<Board chanceLightStartsOn={ false } />);
    expect(asFragment()).toMatchSnapshot();
});

// check handling cell-clicking
it("works when click on a cell", function() {
    const { queryByTestId } = render(<Board chanceLightStartsOn={ false } />);
  
    expect(queryByTestId("0-0")).toBeInTheDocument();
    expect(queryByTestId("0-0")).not.toHaveClass("Cell-lit");
  
    // click on cell coord "1-1"
    const clickCell = queryByTestId("1-1");
    fireEvent.click(clickCell);
    
    expect(queryByTestId("0-1")).toHaveClass("Cell-lit");
    expect(queryByTestId("1-0")).toHaveClass("Cell-lit");
    expect(queryByTestId("1-1")).toHaveClass("Cell-lit");
    expect(queryByTestId("1-2")).toHaveClass("Cell-lit");
    expect(queryByTestId("2-1")).toHaveClass("Cell-lit");
});