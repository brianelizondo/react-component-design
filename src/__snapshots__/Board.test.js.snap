// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'Board' matches snapshot with light start off 1`] = `
<DocumentFragment>
  <div
    class="Board"
  >
    <table>
      <tbody>
        <tr>
          <td
            class="Cell "
            data-testid="0-0"
          />
          <td
            class="Cell "
            data-testid="0-1"
          />
          <td
            class="Cell "
            data-testid="0-2"
          />
        </tr>
        <tr>
          <td
            class="Cell "
            data-testid="1-0"
          />
          <td
            class="Cell "
            data-testid="1-1"
          />
          <td
            class="Cell "
            data-testid="1-2"
          />
        </tr>
        <tr>
          <td
            class="Cell "
            data-testid="2-0"
          />
          <td
            class="Cell "
            data-testid="2-1"
          />
          <td
            class="Cell "
            data-testid="2-2"
          />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
